src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -269,5 +268,56 @@ tf_model_backend::~tf_model_backend() {
   call_pybind_function([&]() { model_ = pybind11::object(); });
 }
 
+tf_image_augmenter::tf_image_augmenter(const options& opts)
+    : processed_image_augmenter(opts) {}
```

Comment:
Could we put the implementation of `tf_image_augmenter` methods up top, just after the definition of the class itself? This way, all the `tf_image_augmenter` stuff is together.

It's also more common idiom in C/C++, to have the file-local helper stuff defined first, followed by the stuff using the helper stuff